### DISCUSSION TITLE
how can I import the new csv of dkb

### DISCUSSION THREAD

**Discussion by Toaster2-0** [firefly-iii/firefly-iii#9436] [Firefly III]
Hello, 

in the import configurations repo is a configuration file for the dkb import, but that doesn't work. Because the csv has three fields relevant to the accounts it needs to be attributed to, first the iban of the other relevant account of the transaction, then the payer (either my dkb account or the other account involved in the transaction) and the payee (again either my dkb account or the other account). If I am using the example configuration for dkb of the import configurations repo it transfers from or to another asset account (if the other asset accounts iban is involved)  and creates a new account for expense or revenue or it throws an error with couldnt find a valid source account. 
I couldn't think of a configuration that resolves my problem, the only option I could think of is to write a script that checks if the amount is negative or positive and depending on that it writes the payer name or the payee name in a new column. 
If you got a better option pls let me know. 

Thanks for helping me in advance.

**Comment by JC5** [comment]
Could you send me some sample lines at james@firefly-iii.org?

**Comment by JC5** [comment]
Alright, thanks for sending me the file.

Since they add extra headers to the file, you must delete those first yourself. It looks like the first four rules must be removed.

Try the following configuration. It works for me ™️ 

```
{
    "version": 3,
    "source": "ff3-importer-develop\/2024-11-05",
    "created_at": "2024-11-10T08:42:02+01:00",
    "date": "d.m.y",
    "default_account": 1,
    "delimiter": "semicolon",
    "headers": true,
    "rules": true,
    "skip_form": false,
    "add_import_tag": true,
    "roles": [
        "date_book",
        "date_transaction",
        "_ignore",
        "_ignore",
        "opposing-name",
        "description",
        "_ignore",
        "opposing-iban",
        "amount",
        "external-id",
        "sepa_db",
        "sepa_ci"
    ],
    "do_mapping": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": [],
    "duplicate_detection_method": "classic",
    "ignore_duplicate_lines": true,
    "unique_column_index": 0,
    "unique_column_type": "internal_reference",
    "flow": "file",
    "content_type": "csv",
    "custom_tag": "",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": false,
    "accounts": [],
    "date_range": "",
    "date_range_number": 30,
    "date_range_unit": "d",
    "date_not_before": "",
    "date_not_after": "",
    "nordigen_country": "",
    "nordigen_bank": "",
    "nordigen_requisitions": [],
    "nordigen_max_days": "90",
    "conversion": true,
    "ignore_duplicate_transactions": true
}
```

**Reply by Toaster2-0** [reply]
Sadly the configuration doesnt work for me, because the name the revenue account gets is wrong if I receive money. If the amount (column index 8) is negative (I transfer money) the column with the index 4 (payee) and 7 (iban) belong together (and my name is in the col with index 3), but when I receive the money the column with the index 3 (payer) and 7 (iban) belong together (and my name is in index 4). So the column with index 7 is the iban of the payer if the amount is positiv, but if the amount is negative the iban (index 7) belongs to the payee.

**Reply by JC5** [reply]
If you have the correct IBANs stored in Firefly III, it should map to those values.

**Reply by Toaster2-0** [reply]
Well yeah if an account for the iban exists, but if not, an account gets created with an iban and name which do not match.

**Reply by mh81** [reply]
Hi @Toaster2-0 ,

I was facing the same problem and asked ChatGPT to create a Python script that would walk through the csv export file and swap the "Zahlungspflichtiger" and "Zahlungsempänger" fields if the the transaction was a debit ("Eingang"). Quick and dirty fix but worked for me.

See script [here](https://gist.github.com/mh81/f66590805bffc86fd3f1d74496e882f5) 

I removed all leading lines in the CSV except the headers before applying the script.

**Reply by Emi1225S** [reply]
This seems to be the working solution. Also, in the [section](https://github.com/firefly-iii/import-configurations/tree/main/de/dkb) for importing dkb csvs, there is a [repo ](https://github.com/MadWalnut/firefly-iii-dkb-csv-fix) linked for fixing dkb csvs. I tried downloading the zip via Brave, which aborted the download due to detecting a virus. If this is an error, i'd suggest putting a warning about it, if not i'd suggest removing that link.
